src/lib.rs

```diff
@@ -293,19 +299,35 @@ pub trait SharedLibrary: Sized + Debug {
     /// Get the debug-id of this shared library if available.
     fn id(&self) -> Option<SharedLibraryId>;
 
-    /// Returns the address of where the library is loaded.
+    /// Returns the address of where the library is loaded into virtual
+    /// memory.
     ///
-    /// This typically is the stated virtual memory address of the first
-    /// code segment but not always.  We follow the breakpad semantics for
-    /// what is considered the load address here so that this information
-    /// can be used to drive server side symbolication systems which are
-    /// generally working with the breakpad definition of what the image
-    /// load address is.
+    /// This address maps to the `Avma` of the first segment loaded into
+    /// memory. Depending on the platform, this segment may not contain code.
```

Comment:
Do either of these still match what breakpad does? If not, do we still need to provide that, since that was the original intent of `load_addr`? (#40)

Reply:
The load address method still match what breakpad does functionally. However, breakpad reads `/proc/self/maps` on Linux to load these values. This would be the best way to implement it, but require much more refactoring than this. I'm not sure how the kernel establishes the size, but we're getting close enough with it. I'll run one more manual test and report back.

On macOS, we're essentially equivalent. Breakpad returns after the first `__TEXT` segment while we would theoretically scan for multiple ones. In practice, I'm pretty sure there only ever is one. 

> I guess this replaces #44 too?

Correct, let me update the PR description.

Reply:
Verified that the number do look correct and that symbolication works with these addresses. We're using it here, although that doesn't show the `addr2line` part of it: 

https://github.com/getsentry/sentry-rust/blob/67f3b24c15da5849d3973f1dfdb36ef2ac7365c7/sentry-debug-images/src/unix.rs#L42-L71

Thanks for double-checking! I hope to get some time eventually to implement a `/proc` based linux implementation.